File: python/179largestNumber/main.py
from typing import List


def largestNumber(nums: List[int]) -> str:
    if all([item == 0 for item in nums]):
        return '0'
    from functools import cmp_to_key
    str_list = []
    for item in nums:
        str_list.append(str(item))

    def compare(a, b):
        c = a + b
        d = b + a
        if c > d:
            return 1
        elif c == d:
            return 0
        else:
            return -1

    str_list.sort(key=cmp_to_key(compare), reverse=True)

    return ''.join(str_list)


print(largestNumber([10, 2]))

File: python/179largestNumber/main.java
    public String largestNumber(int[] nums) {
    if (nums == null || nums.length == 0) {
        return "";
    }
    // 将每个数字转换成字符串
    String[] strs = new String[nums.length];
    for (int i = 0; i < nums.length; i++) {
        strs[i] = String.valueOf(nums[i]);
    }
    // 自定义比较规则
    Arrays.sort(strs, new Comparator<String>() {
        public int compare(String s1, String s2) {
            String order1 = s1 + s2;
            String order2 = s2 + s1;
            return order2.compareTo(order1);
        }
    });
    // 特殊情况处理
    if (strs[0].equals("0")) {
        return "0";
    }
    // 将排序后的字符串连接起来
    StringBuilder sb = new StringBuilder();
    for (String str : strs) {
        sb.append(str);
    }
    return sb.toString();
    }